observation_portal/requestgroups/serializers.py

```diff
@@ -421,6 +452,44 @@ def validate(self, data):
                             ))
                         )
 
+            # Validate the exposure modes
+            if 'exposure' in modes:
+                exposure_mode_validation_helper = ModeValidationHelper('exposure', instrument_type, default_modes,
+                                                                       modes)
+                if exposure_mode_validation_helper.mode_is_not_set(instrument_config['extra_params']):
+                    exposure_mode_to_set = exposure_mode_validation_helper.get_mode_to_set()
+                    if exposure_mode_to_set['error']:
+                        raise serializers.ValidationError(_(exposure_mode_to_set['error']))
+                    if exposure_mode_to_set['mode']:
+                        instrument_config['extra_params']['exposure_mode'] = exposure_mode_to_set['mode']['code']
+
+                exposure_mode_error_msg = exposure_mode_validation_helper.get_mode_error_msg(
+                    instrument_config['extra_params']['exposure_mode'],
+                    instrument_config
+                )
+                if exposure_mode_error_msg:
+                    raise serializers.ValidationError(_(exposure_mode_error_msg))
+
+            # Validate the exposure times for MUSCAT are positive numbers
+            if instrument_type.upper() == '2M0-SCICAM-MUSCAT':
+                exposure_fields = ['exposure_time_g', 'exposure_time_r', 'exposure_time_i', 'exposure_time_z']
+                for exposure_field in exposure_fields:
+                    try:
+                        exposure_time = float(instrument_config['extra_params'][exposure_field])
+                        if exposure_time < 0:
+                            raise serializers.ValidationError(_(f'Muscat instrument extra_param {exposure_field} must be a positive number.'))
+                    except ValueError:
+                        raise serializers.ValidationError(_(f'Muscat instrument extra_param {exposure_field} must be a positive number.'))
+            # This applies the exposure_time requirement only to non-muscat instruments.
+            # I wish I could figure out a better way to do this, but it needs info about the instrument_type which is a level up.
```

Comment:
I don't quite understand - I think `instrument_type` is actually available from here. What other instrument_type info does this section need?

Reply:
sorry, I meant that to validate the exposure time extra params, we need to know that the instrument type is MUSCAT, which is only available at the Configuration level. So even though it would be better to validate the Instrument Config extra params at that level/serializer, we have to do it in the Configuration one because that's where we have the instrument type

Reply:
Ah, gotcha. I also wish there was a better way to do this, but nothing is coming to me right away.